Helm quick-start guide for operator, turtles and rancher (#12)
rancher-sandbox/rancher-turtles-docs: docs/getting-started/install_capi_operator.md

### Install as a `Rancher Turtles` dependency

See the `Rancher Turtles` section for installing operator as a Helm [dependency](./install_turtles_operator.md#install-cluster-api-operator-as-a-helm-dependency)

### Install with Helm
To install `Cluster API Operator` with version `1.4.4` of the `CAPI` + `Docker` provider using helm, follow these steps:

1. Add the Helm repository for the `Cluster API Operator` by running the following command:
```bash
helm repo add capi-operator https://kubernetes-sigs.github.io/cluster-api-operator
```
2. Update the Helm repository by running the following command:
```bash
helm repo update
```
3. Install the `Cluster API Operator` using the following command, which will also install `cert-manager`:
```bash
helm install capi-operator capi-operator/cluster-api-operator
	--create-namespace -n capi-operator-system
	--set infrastructure=docker:v1.4.4
	--set cert-manager.enabled=true
	--timeout 90s --wait # Core Cluster API with kubeadm bootstrap and control plane providers will also be installed
```

> *Note: `cert-manager` is a hard requirement for CAPI and `Cluster API Operator`*

Reviewer: Please add a note, that the cert-manager version is pinned and users can install cert-manager on their own if they prefer

Author: It is already present in the prerequisites

> To provide additional environment variables, choose some feature gates, or provide cloud credentials, similar to `clusterctl` [common provider](https://cluster-api.sigs.k8s.io/user/quick-start#initialization-for-common-providers), in `Cluster API Operator`, a variables secret could be used. A `name` and a `namespace` of the secret could be specified for the `Cluster API Operator`.

Reviewer: We can also include that github token better be included in the secret, it's documented here https://github.com/kubernetes-sigs/cluster-api-operator/tree/main/docs#installing-azure-infrastructure-provider

sidebar_position: 3
---

# Install Cluster API Operator

This section describes how to install `Cluster API Operator` in the Kubernetes cluster.

> ## Installing CAPI and providers

Reviewer 2: ```suggestion
## Installing Cluster API (CAPI) and providers
```

`CAPI` and desired `CAPI` providers could be installed using the helm-based installation for [`Cluster API Operator`](https://github.com/kubernetes-sigs/cluster-api-operator) or as a helm dependency for the `Rancher Turtles`.

### Install as a `Rancher Turtles` dependency (recommended)

See the `Rancher Turtles` section for installing the operator as a Helm [dependency](./install_turtles_operator.md#install-rancher-turtles-operator-with-cluster-api-operator-as-a-helm-dependency)

### Install manually with Helm (alternative)

> To install `Cluster API Operator` with version `1.4.6` of the `CAPI` + `Docker` provider using helm, follow these steps:

Reviewer 2: I think it is worth mentioning why we are using specific CAPI version, v1.4.6 in this case and later versions of it can be used in the future due to some limitations of the Rancher ?

1. Add the Helm repository for the `Cluster API Operator` by running the following command:
```bash
helm repo add capi-operator https://kubernetes-sigs.github.io/cluster-api-operator
```

> 2. Update the Helm repository by running the following command:

Reviewer 2: ```suggestion
2. Update the Helm repository:
```

```bash
helm repo update
```

> 3. Install the `Cluster API Operator` using the following command, which will also install `cert-manager`:

Reviewer 2: ```suggestion
3. Install the `Cluster API Operator`, which will also install `cert-manager`:
```

```bash
helm install capi-operator capi-operator/cluster-api-operator
	--create-namespace -n capi-operator-system
	--set infrastructure=docker:v1.4.6
	--set core=cluster-api:v1.4.6
	--set cert-manager.enabled=true
	--timeout 90s --wait # Core Cluster API with kubeadm bootstrap and control plane providers will also be installed
```

> *Note: `cert-manager` is a hard requirement for CAPI and `Cluster API Operator`*

Reviewer 2: ```suggestion
*Note: `cert-manager` is a hard requirement for `CAPI`and `Cluster API Operator`*
```

> To provide additional environment variables, choose some feature gates, or provide cloud credentials, similar to `clusterctl` [common provider](https://cluster-api.sigs.k8s.io/user/quick-start#initialization-for-common-providers), in `Cluster API Operator`, a variables secret could be used. A `name` and a `namespace` of the secret could be specified for the `Cluster API Operator`.

Reviewer 2: ```suggestion
To provide additional environment variables, enable feature gates, or supply cloud credentials, similar to `clusterctl` [common provider](https://cluster-api.sigs.k8s.io/user/quick-start#initialization-for-common-providers), variables secret with `name` and a `namespace` of the secret could be specified for the `Cluster API Operator` as shown below.
```

```bash
helm install capi-operator capi-operator/cluster-api-operator
	--create-namespace -n capi-operator-system
	--set infrastructure=docker:v1.4.6
	--set core=cluster-api:v1.4.6
	--set cert-manager.enabled=true
	--timeout 90s
	--secret-name <secret_name>
	--secret-namespace <secret_namespace>
	--wait
```

Example secret data:
```yaml
apiVersion: v1
kind: Secret
metadata:
  name: variables
  namespace: default
type: Opaque
stringData:
  CLUSTER_TOPOLOGY: "true"
  EXP_CLUSTER_RESOURCE_SET: "true"
```

To select more than one desired provider to be installed together with the `Cluster API Operator`, the `--infrastructure` flag can be specified with multiple provider names separated by a comma. For example:

```bash
helm install ... --set infrastructure="docker:v1.4.6;azure:v1.4.6"
```

> The `infrastructure` flag is set to `docker:v1.4.6;azure:v1.4.6`, representing the desired provider names. This means that the `Cluster API Operator` will install and manage multiple provider systems, `Docker` and `Azure` respectively, with versions `1.4.6` specified.

Reviewer 2: ```suggestion
The `infrastructure` flag is set to `docker:v1.4.6;azure:v1.4.6`, representing the desired provider names. This means that the `Cluster API Operator` will install and manage multiple providers, `Docker` and `Azure` respectively, with versions `v1.4.6` specified in this example.
```

sidebar_position: 3
---

# Install Cluster API Operator

## Installing CAPI and providers

`CAPI` and desired `CAPI` providers could be installed using helm based installation for [`Cluster API Operator`](https://github.com/kubernetes-sigs/cluster-api-operator).

> ### Cluster API operator path

Author: It was missing reference to the other doc in the hierarchy.

# Install CAPI Operator

## Installing CAPI and providers

CAPI and desired CAPI providers could be installed using helm based installation for [`Cluster API Operator`](https://github.com/kubernetes-sigs/cluster-api-operator).

### Cluster API operator path

> For example, to install latest version of CAPI + docker provider, first, the helm repo should be added with:

Reviewer 3: At this stage we don't want the user to install the latest.

```bash
helm repo add capi-operator https://kubernetes-sigs.github.io/cluster-api-operator
helm repo update
```
and installed together with [cert-manager](https://github.com/cert-manager/cert-manager) using:
```bash

> helm install capi-operator capi-operator/cluster-api-operator

Reviewer 3: Could we include the **set** to make sure v1.4.4 of the core provider is installed.

Reviewer: Depending on that https://github.com/rancher-sandbox/rancher-turtles/pull/109, some values might change and there is no need to install CAPI Operator manually, it's included in extension chart

Author: I’ve updated everything accordingly as https://github.com/rancher-sandbox/rancher-turtles/pull/109 was merged.

# Install CAPI Operator

## Installing CAPI and providers

> There are two options when it comes to installing [CAPI](https://github.com/kubernetes-sigs/cluster-api). You can install CAPI and desired providers with [clusterctl](https://cluster-api.sigs.k8s.io/user/quick-start#install-clusterctl), following the official [quick-start](https://cluster-api.sigs.k8s.io/user/quick-start#installation) guide, or you can use helm based installation with [Cluster API Operator](https://github.com/kubernetes-sigs/cluster-api-operator).

Reviewer 3: As this section is installed CAPI operator then i think we can just cover that and not mention the clusterctl path at all

> There are two options when it comes to installing [CAPI](https://github.com/kubernetes-sigs/cluster-api). You can install CAPI and desired providers with [clusterctl](https://cluster-api.sigs.k8s.io/user/quick-start#install-clusterctl), following the official [quick-start](https://cluster-api.sigs.k8s.io/user/quick-start#installation) guide, or you can use helm based installation with [Cluster API Operator](https://github.com/kubernetes-sigs/cluster-api-operator).

Reviewer 2: Let's avoid using first-person pronounces like "you/we/I" in this repo since it helps a lot to keep them formal

Author: Added some “dryness” to it

```bash
helm install capi-operator capi-operator/cluster-api-operator
	--create-namespace -n capi-operator-system 
	--set infrastructure=docker
	--set cert-manager.enabled=true
	--timeout 90s
	--secret-name <secret_name>
	--secret-namespace <secret_namespace>
	--wait
```

Example secret data:
```yaml
apiVersion: v1
kind: Secret
metadata:
  name: variables
  namespace: default
type: Opaque
stringData:
  CLUSTER_TOPOLOGY: "true"
  EXP_CLUSTER_RESOURCE_SET: "true"
```

You can select providers to install together with Cluster API Operator. To do so, specify one or more providers in `—-infrastructure` flag when starting with helm. You can pass a specific version, if needed:
```bash
helm install ... infrastructure=docker:v1.5.0,azure # For azure assumed latest version
```

> 

Author: Added a link to the document describing core provider installation in more details.
